src/PKSim.Core/Model/RandomPopulationFactory.cs

```diff
@@ -105,21 +121,28 @@ public Task<RandomPopulation> CreateFor(RandomPopulationSettings populationSetti
                      throw new CannotCreatePopulationWithConstraintsException(_reportGenerator.StringReportFor(populationSettings));
 
                   //create a new individual based on population settings defined by the user
-                  updateCurrentIndividualFromSettings(populationSettings, currentIndividual, allDistributedParameters, allBaseDistributedParameters, currentGender, randomPopulation.RandomGenerator);
+                  updateCurrentIndividualFromSettings(populationSettings, baseIndividual, currentIndividual, allDistributedParameters, allBaseDistributedParameters, currentGender, randomPopulation.RandomGenerator);
 
-                  bool success = tryRandomize(currentIndividual, populationSettings, allCurrentIndividualParameters, randomPopulation.RandomGenerator);
-                  if (!success) continue;
+                  var success = tryRandomize(currentIndividual, populationSettings, allCurrentIndividualParameters, randomPopulation.RandomGenerator);
+                  if (!success)
+                     continue;
+
+                  success = diseaseStateImplementation.ApplyForPopulationTo(currentIndividual);
+                  if (!success)
+                     continue;
 
                   randomPopulation.AddIndividualValues(_individualValuesMapper.MapFrom(currentIndividual, allChangedByCreatedIndividualParameters));
 
+                  diseaseStateImplementation.ResetParametersAfterPopulationIteration(currentIndividual);
```

Comment:
For instance GFR_Spec is changed but needs to be set back to a formula

Reply:
>For instance GFR_Spec is changed but needs to be set back to a formula 

Why? And what happens if in my baseIndividual GFR_spec was ovewritten with a value by the user?
At the moment (with healthy populations): if we overwrite a GFR_spec in the individual and create a population based on it: all individuals in the pop get this GFR_spec value entered by user.

Reply:
It needs to be reset to a formula because otherwise there is no way to get the target eGFR. it simply does not make sense to change those parameters if we are using CKD.